src/main/java/org/mskcc/limsrest/service/GetQcReportSamplesTask.java

```diff
@@ -179,6 +180,31 @@ protected void getQcSamples(QcReportSampleList rsl, String dataType) throws Exce
         }
     }
 
+    // COVID19 samples can be linked to a request by their OtherSampleId being the concatenation of investigatorId and iLabServiceId
+    protected List<CovidSample> getCovidSamples(String dataType) throws Exception {
+        List<CovidSample> covidSamples = new ArrayList<>();
+        DataRecord requestRecord = dataRecordManager.queryDataRecords("Request", "RequestId = '" + requestId + "'", this.user).get(0);
```

Comment:
If requestId is invalid and the list is empty, can throw an exception like, "Invalid Id"?

Reply:
Yes, I'd put that in execute()
But I don't think we generally check for whitelist matches and required parameters in the tasks, do we? I think I've only seen this in the controllers.